space-concordia-robotics/robotics-prototype
# Fix commit message hook 363

# Assignee Section

## Description
Fixed the issue with the git commit message hook applying the wrong number from the branch name to the commit message. I didn't bother with making separate unit tests for this hook, as all of the input/output functions used are already tested.

- [x] Slightly refactored branch name verification hook to avoid reusing code.
- [x] Added new Python util function for writing text in files at a given location, used for applying the issue number to the commit message in '.git/COMMIT_EDITMSG'.
- [x] Edited readme with new name of commit hook.
- [x] Add new test case for `get_issue_num` function to ensure that it can properly get the issue number from 'setup-tx2-initialization-275'( the branch where this issue was discovered ).

### Steps for Testing
1) After checking out `fix-commit-message-hook-363`, run `cp commit_message_hook.py .git/hooks/prepare-commit-msg` to setup the new commit hook.
2) While still on 'fix-commit-message-hook-363', temporarily change the name of the branch to one that would fail before the changes in this PR, like `test-123-456` using `git branch -m test-123-456`.
3) Create a dummy file, add it to the staging area and commit it without pushing it.
4) Use `git log` to ensure that the number at of the end of the branch name, was put at the beginning of the commit message.
5) Run `git reset HEAD^` to undo the last commit, the dummy file should be automatically deleted.
6) Change the name of the branch back to the original with `git branch -m fix-commit-message-hook-363`
7) Run `cp commit-message-hook.sh .git/hooks/prepare-commit-msg` on master because the new hook will only work on this branch.

Running this on other branches will not work until if you bring the changes to `utils.py` and `branch_name_verification_hook.py` to those branches. I tested this on 'setup-tx2-initialization-275' and it does fix the issue that the old hook had with that branch name.

closes #363

The approval from all software team leads is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
